Assets/GameAssets/World/WorldScripts/Interactables/AmmoRackInteractable.cs

```diff
@@ -0,0 +1,55 @@
+/**********************************************************************************************************************
+// File Name :         AmmoRackInteractable.cs
+// Author :            Andrew Stapay
+// Creation Date :     11/13/2024
+//
+// Brief Description : Implements an ammo rack where the player can refill their ammo.
+**********************************************************************************************************************/
+using System.Collections;
+using System.Collections.Generic;
+using UnityEngine;
+
+/// <summary>
+/// A class that implements the interaction with the ammo rack
+/// </summary>
+public class AmmoRackInteractable : MonoBehaviour, IPlayerInteractable
+{
+    // The nuumber of harpoons that are currently on the rack
+    private int _currentHarpoons;
+
+    /// <summary>
+    /// Called when the game starts
+    /// Sets the number of harpoons to the number of game objects in the ammo rack in-engine
+    /// </summary>
+    private void Awake()
+    {
+        _currentHarpoons = transform.GetChild(0).childCount;
```

Comment:
This seems odd to me. I feel it would make more sense to input the number of harpoons and have it spawn that many in the barrel rather than the other way arround

Reply:
I thought so too, but the Jira task states that the number of harpoons that are childed to the ammo rack should determine how many ammo pieces you can get from the rack. I'm just doing what was requested of me.

Reply:
Fair enough